src/backend/sgx/ioctls.rs

```diff
@@ -102,3 +108,58 @@ impl<'a> SetAttribute<'a> {
         SetAttribute(fd.as_raw_fd() as _, PhantomData)
     }
 }
+
+#[repr(C)]
+#[derive(Debug)]
+/// SGX_IOC_ENCLAVE_RELAX_PERMISSIONS parameter structure
+pub struct RelaxPermissions<'a> {
+    /// In: starting page offset
+    offset: u64,
+    /// In: length of the address range (multiple of the page size)
+    length: u64,
+    /// In: SECINFO containing the relaxed permissions
+    secinfo: u64,
+    /// Out: length of the address range successfully changed
+    count: u64,
```

Comment:
@jarkkojs If this ioctl does both input and output, shouldn't it be `_IOWR()` and therefore `WriteRead`?

Reply:
Yes, it is `_IOWR()`, you're right.

Reply:
For reference https://lore.kernel.org/linux-sgx/f0029cda125be9eb4eb2e051ab41d500081c6e2c.1644274683.git.reinette.chatre@intel.com/